src/main/java/com/yellow/ordermanageryellow/Service/ProductCategoryService.java

```diff
@@ -4,18 +4,58 @@
 import com.yellow.ordermanageryellow.model.Users;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.CommandLineRunner;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+import org.springframework.http.HttpStatus;
+import java.util.List;
+import java.util.Optional;
 
 @Service
-public class ProductCategoryService implements CommandLineRunner {
+public class ProductCategoryService  {
     private final ProductCategoryRepository ProductCategoryRepository;
     @Autowired
     public ProductCategoryService(ProductCategoryRepository ProductCategoryRepository) {
         this.ProductCategoryRepository = ProductCategoryRepository;
     }
-    @Override
-    public void run(String... args) {
-        ProductCategory ProductCategory = new ProductCategory("12");
-        ProductCategoryRepository.save(ProductCategory);
+    //@Override
+    //public void run(String... args) {
+       // ProductCategory ProductCategory = new ProductCategory("12");
+    //  ProductCategoryRepository.save(ProductCategory);
+   // }
+    public ResponseEntity<List<ProductCategory>> findAll(){
+        List<ProductCategory> categories = this.ProductCategoryRepository.findAll();
+        return new ResponseEntity<>(categories, HttpStatus.OK);    }
+
+    public ResponseEntity<String> insert(ProductCategory newCategory) {
```

Comment:
you should save the product category with comapnay from token

Reply:
I wrote you an email, with some questions and comments
For now I did what I thought
so you can see here and read my email...